File: PN22.cpp


#include<iostream>
using namespace std;


class Number
{
    private:
        int iNo; 
        int iCnt = 0;
		bool bFlag = true;
		

    public:
        
        void Accept()       
        {
            cout<<"Enter the value : "<<endl;
            cin>>this->iNo;
        }

        bool Display()      
        {
            cout<<"Value is : "<<this->iNo<<endl;
        }

        bool CheckPrime()
		{
			
			int iSum = 0;
			
			for(iCnt = 2;iCnt <= (iNo/2); iCnt++)
			{
				if(iNo % iCnt == 0)
				{
					bFlag = false;
					break;
				}
			}
	        return bFlag;
		}
};

int main()
{
    Number nobj;
	bool bRet;

    nobj.Accept();
    nobj.Display();

    bRet = nobj.CheckPrime();
	
	if(bRet == true)
	{
		cout<<"It is prime number: ";
	}
	else
	{
		cout<<"It is not prime number: ";
	}

    return 0;
}

File: PN22.java
// Template for probplems on Numbers

import java.lang.*;
import java.util.*;

class Number
{
    private int iNo;

    public void Accept()
    {
        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter number : ");
        this.iNo = sobj.nextInt();
    }

    public void Display()
    {
        System.out.println("Value is : "+this.iNo);
    }
    
	public int SumDigits()
	{
     int iDigit = 0;
	 int iSum = 0;
	
	 if(iNo < 0)
	 {
		iNo = -iNo;
	 }
	 while(iNo > 0)
	 {
		iDigit = iNo % 10;
		iSum = iSum + iDigit;
		iNo = iNo/10;
	 }
	 return iSum;
	}
}

class PN22
{
    public static void main(String b[])
    {
            Number nobj = new Number();
            int iRet = 0;
			
            nobj.Accept();
            nobj.Display();

            iRet = nobj.SumDigits();
			
			System.out.println("The count of Digits is : "+iRet);
    }
}